lldb/test/Shell/ScriptInterpreter/Python/sb_function_ranges.s

```diff
@@ -0,0 +1,182 @@
+# REQUIRES: x86
```

Comment:
Is there any way we can have an architecture-independent test? If it's a ton of work, I say don't worry about it, most folks are going to build the x86 target probably.

Reply:
The (only?) easy way would be to use a compiler to generate that, but I don't like the tradeoffs that come with that: the `-fbasic-block-sections` is not supported for all (or even most) targets. Most notably, it only supports ELF targets right now. This means that the test would come with a lot of other REQUIRES clauses -- that the developer might not be able do anything about (enabling the x86 target -- if it isn't already -- is much easier than trying to find a machine with a specific os/arch combination).

Using the compiler to generate the input also reduces our control over it, which means any assertion would have to be fairly loose to avoid the test breaking with compiler changes (we definitely couldn't assert the exact ranges, and even checking their count might too brittle).